xtext/org.icyphy.linguafranca/src/org/icyphy/generator/TypeScriptGenerator.xtend

```diff
@@ -236,6 +283,36 @@ class TypeScriptGenerator extends GeneratorBase {
                 println("SUCCESS (compiling generated TypeScript code)")
             }
         }
+        
+        
+        // If this is a federated execution, generate the C RTI
+        
+        // FIXME: DO THE COMMENT BELOW
+        // Also, create two RTI C files, one that launches the federates
+        // and one that does not.
+        
+        if (federates.length > 1) {
+            
+            // Create C output directories (if they don't exist)            
+            dir = new File(cSrcGenPath)
+            if (!dir.exists()) dir.mkdirs()
+            dir = new File(cOutPath)
+            if (!dir.exists()) dir.mkdirs()
+            
+            generateFederateRTI()
+
+            // Copy the required library files into the target file system.
+            // This will overwrite previous versions.
+            var files = newArrayList("rti.c", "rti.h", "federate.c", "reactor_threaded.c", "reactor.c", "reactor_common.c", "reactor.h", "pqueue.c", "pqueue.h", "util.h", "util.c")
+
+            for (file : files) {
+                copyFileFromClassPath(
+                    File.separator + "lib" + File.separator + "C" + File.separator + file,
+                    cSrcGenPath + File.separator + file
+                )
+            }
+            compileRTI()
```

Comment:
This should be handled in a different class. All TypeScriptGenerator should care about is that an RTI gets produced. It should not have to deal with it's dependencies or compilation details. This information should not be scattered around the code base.